pkg/cloud_provider/auth/token_sources.go

```diff
@@ -119,6 +119,11 @@ func (ts *GCPTokenSource) fetchIdentityBindingToken(ctx context.Context, k8sSATo
 		ts.meta.GetIdentityPool(),
 		ts.meta.GetIdentityProvider(),
 	)
```

Comment:
If we wanted to support hostnetwork, I think we would also need to make this custom audience change to getAudienceFromContextAndIdentityProvider [here](https://github.com/GoogleCloudPlatform/gcs-fuse-csi-driver/blob/072dfe3f4655b65ac19d2bbb8d49a73731a42402/pkg/sidecar_mounter/sidecar_mounter.go#L359-L363)  which is used by hostnetwork.